include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -6096,7 +6185,109 @@ struct OrtCompileApi {
 
 ORT_RUNTIME_CLASS(Ep);
 ORT_RUNTIME_CLASS(EpFactory);
+ORT_RUNTIME_CLASS(MemoryDevice);  // opaque class to wrap onnxruntime::OrtDevice
```

Comment:
I couldn't avoid adding OrtDevice to the API as IDataTransfer uses it. To try and disambiguate/clarify I called it OrtMemoryDevice as we have OrtHardwareDevice, and calling it `OrtDevice` in the API felt too vague.